lib/spack/llnl/util/filesystem.py

```diff
@@ -1133,21 +1135,31 @@ def lexists_islink_isdir(path):
     return True, is_link, is_dir
 
 
+class BaseDirectoryVisitor(object):
+    """Base class and interface for visit_directory_tree."""
```

Comment:
Just wanted to point out that we might use https://python.readthedocs.io/en/latest/library/abc.html#module-abc if it makes sense. Otherwise, this is fine coded as a normal base class.

Reply:
I think it doesn't have to be abstract, cause the `return False` is probably sensible for symlinked dirs.